### PR TITLE
Khateeb auth-challenge

### PR DESCRIPTION
**Pull request by Khateeb1111**

### CODE REVIEW

File: src/client/App.jsx [hunk 1]
```diff
@@ -15,10 +15,54 @@ function App() {
...
+        //If the server is unavailable, then show an error
+        //to the user
+        console.log(error)
+      })
     
   };
 
   const handleLogin = async ({ username, password }) => {
+    const options = {
+      //Send a post request
```
Comment (BrandonV21) [inline src/client/App.jsx:46]: I like how you added brief comments to explain what the code is doing

File: src/server/controllers/user.js [hunk 2]
```diff
@@ -8,27 +8,42 @@ const jwtSecret = 'mysecret';
...
+            password: hashedPassword
+          },
+        });
+      
+        res.json({ data: createdUser });
+      };
+
+//     const createdUser = null;
+
+//     res.json({ data: createdUser });
```
Comment (BrandonV21) [inline src/server/controllers/user.js:24]: Best to clean the code up if you don't plan to use it before submitting the work.

Comment (New-Arday) [reply src/server/controllers/user.js:24]: I agree with Brandon's comment about removing the comments when you implement the code for it.

File: src/server/controllers/user.js [hunk 3]
```diff
@@ -8,27 +8,42 @@ const jwtSecret = 'mysecret';
...
 
-    const passwordsMatch = false;
+    const passwordsMatch = await bcrypt.compare(password, foundUser.password);
 
     if (!passwordsMatch) {
         return res.status(401).json({ error: 'Invalid username or password.' });
     }
 
-    const token = null;
+    const token = jwt.sign({username}, jwtSecret);
```
Comment (New-Arday) [inline src/server/controllers/user.js:46]: I'm not sure how you were able to create two const variables with the same name...

**Pull request closed without merging**